openslides/core/static/js/core/site.js

```diff
@@ -342,61 +342,114 @@ angular.module('OpenSlidesApp.core.site', [
     }
 ])
 
-.factory('Multiselect', [
+/* This factory handles the filtering of the OS-data-tables. It contains
+ * all logic needed for the table header filtering. Things to configure:
+ * - multiselectFilters: A dict associating the filter name to a list (empty per default). E.g.
+ *       { tag: [],
+ *         category: [], }
+ * - booleanFilters: A dict containing a dict for every filter. The value property is a must.
+ *   For displaying properties like displayName, choiceYes and choiceNo could be usefull. E.g.
+ *      { isPresent: {
+ *          value: undefined,
+ *          displayName: gettext('Is present'), } }
+ * - propertyList, propertyFunctionList, propertyDict: See function getObjectQueryString
+ */
+.factory('osTableFilter', [
     function () {
-        var instance = function () {
-            var areFiltersSet = function () {
+        var createInstance = function () {
+            var self = {
+                multiselectFilters: {},
+                booleanFilters: {},
+                filterString: '',
+            };
+            self.areFiltersSet = function () {
                 var areFiltersSet = false;
-                _.forEach(self.filters, function (filterList, filter) {
+                _.forEach(self.multiselectFilters, function (filterList, filter) {
                     if (filterList.length > 0) {
                         areFiltersSet = true;
                     }
                 });
+                _.forEach(self.booleanFilters, function (filterDict, filter) {
```

Comment:
Here you can use a `_.find` or something similar that breaks the loop if one truthy value was found.